lib/utils/permission_service.dart

```diff
@@ -46,6 +46,17 @@ class PermissionHandlerService {
     }
   }
 
+  Future<PermissionStatus> requestPermissionForMircoActions() async {
+    final currentStatus = await Permission.microphone.status;
+    if (currentStatus == PermissionStatus.denied ||
+        currentStatus == PermissionStatus.permanentlyDenied) {
+      final newStatus = await Permission.microphone.request();
+      return newStatus.isGranted ? PermissionStatus.granted : newStatus;
```

Comment:
I think you can directly do this here:
`return Permission.microphone.request()`